ogcapi-stable/ogcapi-common/src/main/java/de/ii/ldproxy/ogcapi/infra/rest/OgcApiExceptionMapper.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2020 interactive instruments GmbH
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package de.ii.ldproxy.ogcapi.infra.rest;
+
+import de.ii.xtraplatform.dropwizard.api.Jackson;
+import io.dropwizard.jersey.errors.LoggingExceptionMapper;
+import io.dropwizard.views.View;
+import io.dropwizard.views.mustache.MustacheViewRenderer;
+import org.apache.felix.ipojo.annotations.Component;
+import org.apache.felix.ipojo.annotations.Instantiate;
+import org.apache.felix.ipojo.annotations.Provides;
+import org.apache.felix.ipojo.annotations.Requires;
+import org.apache.felix.ipojo.annotations.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author zahnen
+ */
+@Component
+@Provides
+@Instantiate
+@Provider
+@Produces({MediaType.WILDCARD})
+public class OgcApiExceptionMapper extends LoggingExceptionMapper<Throwable> implements MessageBodyWriter<OgcApiErrorMessage> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OgcApiExceptionMapper.class);
+
+    @Requires
+    Jackson jackson;
+
+    @Context
+    private UriInfo uriInfo;
+
+    @Validate
+    void onStart() {
+        LOGGER.debug("OGC API EXCEPTION WRITER");
+    }
+
+    @Override
+    public Response toResponse(Throwable exception) {
```

Comment:
This looks as if everything that it not a WebApplicationException is mapped to a 500, which is not correct. Some of the exceptions need to be mapped to 400, 404, etc. Or am I missing something?

Reply:
Yes, you're right. So as you suggested below, I will summarize how to map different error scenarios to Java exceptions and how they in turn are mapped to Web exceptions